[LYHTHADDEUS] iP

lyhthaddeus opened the pull request:
# Daiyan

> The sun sets westward, yet the wind carries nary a melody of the string... the gales of war are 
> blowing. I fear that something immense is afoot here. 

Commander, I am here to assist you should you require my assistance. I can help:
* Record your task at hand
* Keep track of task completion progress
* *Quick and Easy* to use

If you think my assistance will be helpful please
1. clone the repo from [here](https://github.com/lyhthaddeus/ip)
2. run the code
3. add the task and enjoy 😄 

The service I provide to you is **free**

Features
- [X] Managing Task
- [ ] Reminder on deadlines

My services are written in Java. `Hello world`
```Java
Public static void main(String[] Args) {
    System.out.print("Hellow World")
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

killmenowya committed:
Completed the task for level 0

lyhthaddeus committed:
modified:   Daiyan.java
	new file:   Logic.java

lyhthaddeus committed:
modified:   Daiyan.java
	modified:   Logic.java

lyhthaddeus committed:
new file:   Command.java
	new file:   CommandList.java
	modified:   Logic.java

lyhthaddeus committed:
modified:   Command.java
	modified:   CommandList.java
	modified:   Logic.java

lyhthaddeus committed:
Completed task for level 4

lyhthaddeus committed:
new file:   src/main/java/Controller/Logic.java
	modified:   src/main/java/Daiyan.java
	renamed:    src/main/java/CommandList.java -> src/main/java/DataStructure/CommandList.java
	deleted:    src/main/java/Deadline.java
	deleted:    src/main/java/Event.java
	new file:   src/main/java/Exception/InvalidInputException.java
	new file:   src/main/java/Exception/SyntaxException.java
	deleted:    src/main/java/Logic.java
	renamed:    src/main/java/Command.java -> src/main/java/TaskObjects/Command.java
	new file:   src/main/java/TaskObjects/Deadline.java
	new file:   src/main/java/TaskObjects/Event.java
	renamed:    src/main/java/Todo.java -> src/main/java/TaskObjects/Todo.java
	modified:   text-ui-test/EXPECTED.TXT
	modified:   text-ui-test/input.txt
	modified:   text-ui-test/runtest.bat

lyhthaddeus committed:
finish level 6 deletion of task

lyhthaddeus committed:
added enums for better oop

lyhthaddeus committed:
changed readme

lyhthaddeus committed:
updated README

lyhthaddeus committed:
commit daiyan

lyhthaddeus committed:
added daiyan img

lyhthaddeus committed:
Add Storage.java and StorageSyntaxException.java

App lacks the ability to Store information from
previous usage.

Each use is isolated from each other and data
is not brought forward

Add a Storage class to handle storage of data
into plain text as well as loading of said data
on start up

New Exception StorageSyntaxException created
to handle potential exception arising from
new Class.

Command refactored to Task in all instance to
better fit description.

lyhthaddeus committed:
Merge branch-Level-7 into master

lyhthaddeus committed:
Implement Level 8 DateTime feature

Deadline and Event class: add LocalDateTime

Both class have been dealing with time as plain string, usage of dateTime
allow for future expandability

Let's add DateTime funcationality to both Deadline and Event class

Adding this allows for much easier and accurate comparison of time
compare to using plain string

lyhthaddeus committed:
Merge branch-Level-8 into master

lyhthaddeus committed:
Add more OOP to code base

Add Ui class and Command package

Parser class was handling all the logic for parsing as
well as Ui. Tasklist was handling Command logic on top
of being a datastructure.

Tasklist and Parser handles too many task which cause
messy monolithic codebase.

Extracting a Ui class and a Command package will help
the codebase be easier to read and follow.

Let's pull up these extra features into its own
unique class.

Refactored main to also handle basic logic flow.

lyhthaddeus committed:
Merge branch 'A-MoreOOP'

lyhthaddeus committed:
Merge branch 'add-gradle-support'

lyhthaddeus committed:
Add Find feature

Ability to filter task by keyword was added

Task list would get hard for users to search through if
list gets too large

Create a feature for filtering task by keyword using
simple String matching. Tasklist has new method find()
that filters through the ArrayList and returns String
of task that matches keyword. FindCommand also implmented
and added to Enum and Parser logic.

Feature was add on the same way other command like add
was implemented.

lyhthaddeus committed:
Add JavaDoc

Add Java Doc for clear documentation for future reference

lyhthaddeus committed:
Fix Coding Style

Fix the coding style of the code base to comply with
coding standards

lyhthaddeus committed:
Merge branch 'branch-A-CodingStandard'

lyhthaddeus committed:
Merge branch-Level-9

Merge branch level 9 with the feature to filter task list

adwinang commented:
Quite well done, and easy understand your code with good folder structuring and naming conventions

popova-mariia commented on src/main/java/Enums/CommandTypes.java:
```diff
@@ -0,0 +1,34 @@
+/**
+ * Enum of all possible commands
+ */
+public enum CommandTypes {
```
I like how you added Enums class given that its optional

popova-mariia commented on src/main/java/Controller/Storage.java:
```diff
@@ -0,0 +1,117 @@
+package Controller;
```
Shouldnt package name be lowercase? I noticed that issue in several other places too.

popova-mariia commented on src/main/java/app/Daiyan.java:
```diff
@@ -0,0 +1,70 @@
+        Parser parser = new Parser(storage);
+
+        this.ui.showWelcomeMessage();
+        ;
```
shouldnt this semicolon be removed?

popova-mariia commented on src/main/java/Commands/ExitCommand.java:
```diff
@@ -0,0 +1,48 @@
+
+    /**
+     * Executes the command by saving the calling storage class to save the
+     * data then call user interface to print goodbye message
```
Shouldnt JavaDocs end each sentence with a fullstop? same goes for @param and @throws. I noticed this issue in several other places too.

popova-mariia reviewed (commented):
Overall great job 😊

izniy commented on src/main/java/Controller/Storage.java:
```diff
@@ -0,0 +1,117 @@
+package Controller;
+
+import DataStructure.TaskList;
+import TaskObjects.*;
```
Imported classes should be listed explicitly instead of using the wildcard import

izniy commented on src/main/java/Controller/Ui.java:
```diff
@@ -0,0 +1,65 @@
+     * Prints standard welcome message when starting the app
+     */
+    public void showWelcomeMessage() {
+        String asciiArt = " ____   ____   _ __  __ ____   __  _ \n" +
```
This is cool!

izniy commented on src/test/java/DataStructure/TaskListTest.java:
```diff
@@ -0,0 +1,29 @@
+package DataStructure;
```
Names representing packages should be in all lowercase

izniy reviewed (commented):
LGTM. Just some nits to fix. Good job :)

lyhthaddeus committed:
Add Checkstyle

Add checkstyle to assist in coding style

to ensure a easier time coding and maintaining to conding
style standards, checkstyle helps catches potential
style errors that may slip our checks

automated process of checkstyle helps with speed
of coding

lyhthaddeus committed:
Add JavaFX GUI

Daiyan class: extracted main and JavaFX GUI feature

Previous Ui of text based CLI commands may be hard for
users to parse and use

Creating a simple GUI using JavaFX will greatly make
app more readable to users and looks nicer on
the eyes as well

Let's make a proper GUI for our app insted
of CLI commands

Using JavaFX as its reccomended by lecturers

lyhthaddeus committed:
Add Assertions to code

Previous code did not have assertions and some values
could be pass through as unexpected values during
runtime.

the use of assertions can help alleviate the issue
and ensure the robustness of the code. Preventing
the runtime errors that could occur.

lyhthaddeus committed:
Add assertions

lyhthaddeus committed:
Add Sort feature

lyhthaddeus committed:
Update README.md

lyhthaddeus committed:
Add Ui image for User guide

lyhthaddeus committed:
Update User Guide

lyhthaddeus committed:
Update User Guide

lyhthaddeus committed:
Update README.md

lyhthaddeus committed:
Add Assertions

lyhthaddeus committed:
Merge pull request #2 from lyhthaddeus/branch-A-Assertions

Branch a assertions

lyhthaddeus committed:
Fix Code Quality

lyhthaddeus committed:
Merge branch 'master' into branch-A-CodeQuality

lyhthaddeus committed:
Merge pull request #3 from lyhthaddeus/branch-A-CodeQuality

Fix Code Quality
